src/middleware/graphql.ts

```diff
@@ -124,6 +170,56 @@ export abstract class GraphQLBase implements Initializable<express.Application>
     this.server = this.createServer(schemaWithMiddleware);
   }
 
+  private async updateRemoteSchemaVersion(url: string) {
+    log.info(`Update schema version for ${url} with current version ${this.remoteSchemaVersions[url]}`);
+    const connector = new GraphQLAPIConnector({ customEndpoint: url, logTag: 'Schema Update', urls: {} });
+    try {
+      const response = await connector.query(Context.createEmpty(), `{ version }`);
+      log.info(`Update remote schema version response: ${JSON.stringify(response)}`);
+      this.remoteSchemaVersions[url] = response && response.version;
+    } catch (e) {
+      log.error(`Failed to get remote schema version for ${url}`, e);
+    }
+  }
+
+  private startRetryOperation(url: string, targetVersion?: string) {
+    const shouldRetryForRemoteSchema =
+      this.getRemoteSchemaStatus(url) === RemoteSchemaStatus.INACTIVE ||
+      (Boolean(targetVersion) && this.remoteSchemaVersions[url] !== targetVersion);
+
+    if (shouldRetryForRemoteSchema) {
+      const self = this;
+      const updateAttempt = () => self.updateRemoteSchema(url);
```

Comment:
I'm not sure if this technique is needed here. Would the `this` keyword refer to something else when this lambda is later executed? Would `bind` help here or do it in a clearer way?